### PR TITLE
[mod] hardening "calculator plugin" / limit execution time to 50 ms 

### PR DESCRIPTION
**Pull request by return42**
### [mod] hardening "calculator plugin" / limit execution time to 50 ms

The execution of the function for the calculation is outsourced to a process whose runtime is limited to 50 milliseconds.

### [mod] enable calculator and allow plugin on public instances

Remove quirks that prevented the Calculator from being used on public instances.

### Related:

- https://github.com/searxng/searxng/pull/3377#issuecomment-2067977375

### CODE REVIEW

File: searx/plugins/calculator.py [hunk 1]
```diff
@@ -51,10 +55,31 @@ def _eval(node):
     raise TypeError(node)
 
 
+def timeout_func(timeout, func, *args, **kwargs):
```
Comment (glanham-jr) [inline searx/plugins/calculator.py:58]: This seems like a useful utility method that could be used for any general func. Do you think this has the possibility to be used in other places in the project, and if so for it to belong in a separate shared file?

Comment (return42) [reply searx/plugins/calculator.py:58]: I had already thought about it, but then decided against it: Forking a new process is a resource "expensive" operation and should remain the exception. Should we one day have the need elsewhere, we could then consider moving this function to another module such as `searx.utils`.

File: searx/plugins/calculator.py [hunk 2]
```diff
@@ -51,10 +55,31 @@ def _eval(node):
...
+
+    def handler(q: Queue, func, args, **kwargs):  # pylint:disable=invalid-name)
```
Comment (Bnyro) [inline searx/plugins/calculator.py]:
```suggestion
    def handler(q: Queue, func, args, **kwargs):  # pylint:disable=invalid-name
```
small typo

**Review by Bnyro — APPROVED** [review]
This works as expected and feels like a much better solution than limiting it to public instances only.

Great job!

File: searx/plugins/calculator.py (hunk 2 above)
Comment (return42) [reply searx/plugins/calculator.py]: Thanks! .. fixed / [diff](https://github.com/searxng/searxng/compare/719edaf0daaa31625fc1abf868530bcf63729579..5c1d4710d5d768e67081781a748af3c6fb863c58)

**Pull request merged**